modules/solid_properties/src/materials/SolidPropertiesMaterial.C

```diff
@@ -0,0 +1,85 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SolidPropertiesMaterial.h"
+
+const std::string SolidPropertiesMaterial::_name = "";
+
+InputParameters
+SolidPropertiesMaterial::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription("Base class for defining solid property materials");
+  return params;
+}
+
+SolidPropertiesMaterial::SolidPropertiesMaterial(const InputParameters & parameters)
+  : DerivativeMaterialInterface<Material>(parameters)
+{
+}
+
+const std::string &
+SolidPropertiesMaterial::solidName() const
+{
+  return _name;
+}
+
+Real
+SolidPropertiesMaterial::molarMass() const
+{
+  mooseError(name(), ": ", __PRETTY_FUNCTION__, " not implemented.");
```

Comment:
For all of the methods that follow, remove `name()`. This info cmoes automatically from a `MooseObject` as a prefix:

```
mooseError(__PRETTY_FUNCTION__, " not implemented.");
```

Reply:
Good to know! I'll go through and update the `fluid_properties` module as well, `name()` is used all over the place in that one